repo.json

```
@@ -1,21 +1,45 @@
 {
-  "pkg_format": "debian",
-  "packages": {
-    "name": "termux-main",
-    "distribution": "stable",
-    "component": "main",
-    "url": "https://packages-cf.termux.dev/apt/termux-main"
-  },
-  "root-packages": {
-    "name": "termux-root",
-    "distribution": "root",
-    "component": "stable",
-    "url": "https://packages-cf.termux.dev/apt/termux-root"
-  },
-  "x11-packages": {
-    "name": "termux-x11",
-    "distribution": "x11",
-    "component": "main",
-    "url": "https://packages-cf.termux.dev/apt/termux-x11"
-  }
+    "repo__package_manager": "apt",
+    "channels": {
+        "packages": {
+            "name": "termux-main",
+            "distribution": "stable",
+            "component": "main",
+            "signing_key_names": [ "termux-autobuilds.gpg", "grimler.gpg" ]
+        },
+        "root-packages": {
+            "name": "termux-root",
+            "distribution": "root",
+            "component": "stable",
+            "signing_key_names": [ "termux-autobuilds.gpg", "grimler.gpg" ]
+        },
+        "x11-packages": {
+            "name": "termux-x11",
+            "distribution": "x11",
+            "component": "main",
+            "signing_key_names": [ "termux-autobuilds.gpg", "grimler.gpg" ]
+        },
+        "disabled-packages": {
+            "is_disabled": "true",
```

This could be simply a boolean instead of a string
```suggestion
            "is_disabled": true,
```

---

Its not the packages that are disabled, its the repo channel, forks may have multiple disabled channels, including x11/root. That sets the `TERMUX_REPO__DISABLED_CHANNEL_NAMES` and `TERMUX_REPO__DISABLED_CHANNEL_DIRS` variables.

- https://github.com/termux/termux-packages/blob/fd7d763463e8925d2544aea72629f1861f43a171/scripts/utils/termux/repository/termux_repository.sh#L66-L68
- https://github.com/termux/termux-packages/blob/fd7d763463e8925d2544aea72629f1861f43a171/scripts/utils/termux/repository/termux_repository.sh#L313-L314

---

Welcome. Yeah, normally I would agree with you, positive variables are better. But if the default is `true` and does not need to be specified in a config file, then it is normally okay to have negative variables where they only need to be added to some configs. And in our case we already have `TERMUX_REPO__DISABLED_*` variables (and also a `disabled-packages` repo dir), so would be good to be consistent.